src/loss.jl

```diff
@@ -106,7 +145,7 @@ function cudnnSoftmaxForward{T}(x::KnetArray{T}; algo=0, mode=0, alpha=1, handle
     y = similar(x)
     @cuda(cudnn, cudnnSoftmaxForward,
           (Cptr, Cint, Cint, Ptr{T}, Cptr, Ptr{T}, Ptr{T}, Cptr, Ptr{T}),
-          handle, algo, mode, Ref(T(alpha)), TD4(x), x, Ref(T(beta)), TD4(y), y)
+          handle, algo, mode, Ref(T(alpha)), TD(x), x, Ref(T(beta)), TD(y), y)
```

Comment:
Why did we go back to TD from TD4 here?  cudnn does not like 1-2 dimensional arrays, TD4 reshapes to make cudnn happy.

Reply:
I removed `TD4` because I'm reshaping the inputs before calling `cudnnSoftmaxForward`